### PR TITLE
Restrict access to private API apple_common to the repos and tests that need it.

### PR DESCRIPTION
**Pull request by pzembrod**
In AppleBootstrap.java give several rule repos access to apple_common that need it, and activate the access restriction independent of INCOMPATIBLE_STOP_EXPORTING_LANGUAGE_MODULES.

PiperOrigin-RevId: 696999389
Change-Id: I000911fbef3b48039beb423e5ba1ab48059b8eeb

### CODE REVIEW

**Review by meteorcloudy — APPROVED** [review]

**Comment by meteorcloudy** [comment]
Waiting for @comius 's final approval.

**Review by comius — APPROVED** [review]

**Comment by brentleyjones** [comment]
https://github.com/bazelbuild/bazel/issues/24375

**Pull request merged**
